gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannelTest.java

```diff
@@ -309,6 +309,40 @@ public void writeHandlesErrorOnQueryWriteStatusRequest() throws Exception {
     writeChannel.write(data.asReadOnlyByteBuffer());
   }
 
+  @Test
+  public void writeHandlesErrorOnInsertRequestWithUnnommitedData() throws Exception {
+    GoogleCloudStorageGrpcWriteChannel writeChannel = newWriteChannel();
+    int chunkSize = GoogleCloudStorageGrpcWriteChannel.GCS_MINIMUM_CHUNK_SIZE;
+    fakeService.setInsertRequestException(new IOException("Error!"));
+    fakeService.setQueryWriteStatusResponses(
+        ImmutableList.of(
+            QueryWriteStatusResponse.newBuilder().setCommittedSize(chunkSize * 3 / 4).build())
+            .iterator());
+
+    ByteString data = createTestData(chunkSize * 3 / 2);
+    writeChannel.initialize();
+    writeChannel.write(data.asReadOnlyByteBuffer());
+
+    assertThrows(IOException.class, writeChannel::close);
+  }
+
+  @Test
+  public void writeHandlesErrorOnInsertRequestWithoutUncommitedData() throws Exception {
```

Comment:
Can we add integration tests for this?

Reply:
Seems like this is not an integration test file and we are missing integration tests for gRPC write/read. I will create another PR to draft integration tests for both read/write.

Reply:
ack